packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,182 @@
+      data-testid="checkbox-indeterminate"
+    >
+      <Icon
+        color="colorBackground"
+        decorative={true}
+        icon="MinusSmallIcon"
+        size="sizeIcon30"
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export interface CheckboxProps extends RadixCheckboxProps {
+  /** Wrap or leave checkbox as is. */
+  wrapped?: boolean;
+  /** Provide error state of the checkbox. */
+  error?: boolean;
+  /** Checkbox id is required for proper functioning of htmlFor attribute on label. */
+  checkboxId: string;
+  /** The contents of the anchor. Can be text or valid text related HTML, i.e. strong elements. */
+  children: NonNullable<React.ReactNode>;
+}
+
+const getCheckboxStyles = (
+  wrapped: boolean
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme>;
+  padding: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  if (wrapped) {
+    return {
+      backgroundColor: "colorBackgroundWeak",
+      padding: "space50",
+    };
+  }
+  return {
+    backgroundColor: "colorBackground",
+    padding: "space0",
+  };
+};
+
+const StyledCheckbox = styled.input`
+  width: 16px;
+  height: 16px;
+  border-radius: 4px;
+  margin-right: ${theme.space.space30};
+  padding: 4px 0 0;
+
+  &:focus {
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05), 0 0 0 2px ${
+      theme.colors.colorBackground
+    }, 0 0 0 4px ${theme.colors.colorBackgroundPrimary};
+    outline: none;
+  },
+
+  &[data-state=unchecked] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${(props: { error: boolean }) =>
+      props.error
+        ? theme.colors.colorBackgroundDestructive
+        : theme.colors.colorBorder};
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBorder};
+    },
+  },
+  
+  &[data-state=checked], &[data-state=indeterminate] {
+    background-color: ${theme.colors.colorBackgroundPrimary};
+    border: 1px solid ${theme.colors.colorBackgroundPrimary};
+
+    &[data-disabled] {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    }
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundPrimaryStrong};
+      border: 1px solid ${theme.colors.colorBackgroundPrimaryStrong};
+    },
+  },
+  
+  &[data-disabled] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${theme.colors.colorBackgroundStrong};  
+  
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    },
+  },
+`;
+
+export const Checkbox = React.forwardRef<HTMLButtonElement, CheckboxProps>(
+  (
+    { wrapped = false, error = false, checkboxId, checked, children, ...props },
+    ref
+  ) => {
+    return (
+      <Box.div
+        alignItems="center"
+        borderRadius="borderRadius30"
+        data-testid="checkbox-container"
```

Comment:
We shouldn't use data-testids in Pluto since we can cause issues for projects that will use Pluto and might use the same data-testid. cc @richbachman x)